src/main/java/de/tum/in/www1/artemis/web/rest/lecture/VideoUnitResource.java

```diff
@@ -90,6 +90,10 @@ public ResponseEntity<VideoUnit> updateVideoUnit(@PathVariable Long lectureId, @
 
         // Validate the URL
         try {
+            // Remove leading and trailing whitespaces
+            if (videoUnit.getSource() != null) {
+                videoUnit.setSource(videoUnit.getSource().strip());
+            }
```

Comment:
```suggestion
        normalizeVideoUrl(videoUnit);

        // Validate the URL
        try {
```

Two suggestions:
- The new code should be in a small helper method to avoid duplication. This also makes possible future necessary changes to the URL easier.
- It should not be part of the `try`-block of the validation. (Sidenote: The validation should probably be moved to a separate small method to remove duplication as well).

Reply:
Yes you are right :+1: extracted both duplicated code parts :+1: